Mage.Sets/src/mage/cards/t/TalonOfPain.java

```diff
@@ -0,0 +1,156 @@
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.t;
+
+import java.util.Objects;
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.RemoveVariableCountersSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DamageTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreatureOrPlayer;
+
+/**
+ *
+ * @author jerekwilson
+ */
+public class TalonOfPain extends CardImpl {
+
+    public TalonOfPain(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{4}");
+        
+
+        /*
+         * Whenever a source you control other than Talon of Pain deals damage to an opponent, 
+         * put a charge counter on Talon of Pain.
+         */
+        this.addAbility(new TalonOfPainTriggeredAbility());
+           
+        
+        // {X}, {tap}, Remove X charge counters from Talon of Pain: Talon of Pain deals X damage to target creature or player.
+        Ability ability  = new SimpleActivatedAbility(Zone.BATTLEFIELD, new DamageTargetEffect(new ManacostVariableValue()) , new ManaCostsImpl("{X}"));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new RemoveVariableCountersSourceCost(CounterType.CHARGE.createInstance()));
+        ability.addTarget(new TargetCreatureOrPlayer());
+        this.addAbility(ability);
+        
+        
+    }
+
+    public TalonOfPain(final TalonOfPain card) {
+        super(card);
+    }
+
+    @Override
+    public TalonOfPain copy() {
+        return new TalonOfPain(this);
+    }
+    
+    private class TalonOfPainTriggeredAbility extends TriggeredAbilityImpl {
+
+        public TalonOfPainTriggeredAbility() {
+            super(Zone.BATTLEFIELD, new TalonOfPainEffect());
+        }
```

Comment:
You can just use `new AddCountersSourceEffect(CounterType.CHARGE.createInstance())`